src/include/sof/debug/debug.h

```diff
@@ -138,4 +138,39 @@
 #define dump_object_ptr(__o) do {} while (0)
 #endif
 
+#if CONFIG_TRACE
+/* swap an endiannes of 32-bit word */
+#define byte_swap(N) ({ \
+	typeof(N) n = (N); \
+	((n & 0x000000FF) << 24) | \
+	((n & 0x0000FF00) << 8) | \
+	((n & 0x00FF0000) >> 8) | \
+	((n & 0xFF000000) >> 24); })
+
```

Comment:
Ok, I understand now why you need this. You want to print the data in hexump in big endian so that it can be easily read. This isn't really the way to go because you assume that dump_hex is called on little endian platforms! But it wont work on big endian platforms.

What you want is a function from the family `cpu_to_le32`. But this will take some time to get it right, so I am only suggestion now the following:

1. Rename byte_swap to bsawp32 and place it outside of `CONFIG_TRACE` and later we can add `cpu_to_*` variants.

Reply:
added suggested changes, thank you